src/foam/nanos/crunch/ExpireUserCapabilityJunctionsCron.java

```diff
@@ -28,53 +28,30 @@ public class ExpireUserCapabilityJunctionsCron implements ContextAgent {
 
   @Override
   public void execute(X x) {
-    logger = (Logger) x.get("logger");
+    logger = Loggers.logger(x, this);
     DAO bareUserCapabilityJunctionDAO = (DAO) x.get("bareUserCapabilityJunctionDAO");
     DAO userCapabilityJunctionDAO = (DAO) x.get("userCapabilityJunctionDAO");
-    Date today = new Date();
 
-    // Find all junctions that are past the expiration date and filter by the following
-    //    - If ucj is still GRANTED and not in grace period, it needs to be returned 
-    //      and reput such that it is either in graceperiod (if applicable) or expired
-    //    - if ucj is in grace period, and the graceperiod is less than or equals to 0 days left
-    //    - return the ucj and reput it in EXPIRED status
+    // Find all GRANTED junctions that are past the expiration date and if not in Grace period, expire them.
     List<UserCapabilityJunction> activeJunctions = ((ArraySink) userCapabilityJunctionDAO
       .where(
         AND(
+          EQ(UserCapabilityJunction.STATUS, CapabilityJunctionStatus.GRANTED),
           NEQ(UserCapabilityJunction.EXPIRY, null),
-          LT(UserCapabilityJunction.EXPIRY, today),
-          OR(
-            AND(
-              EQ(UserCapabilityJunction.IS_IN_GRACE_PERIOD, true),
-              LTE(UserCapabilityJunction.GRACE_PERIOD, 0)),
-            AND(
-              EQ(UserCapabilityJunction.IS_IN_GRACE_PERIOD, false),
-              EQ(UserCapabilityJunction.STATUS, CapabilityJunctionStatus.GRANTED))
-          )
-      ))
+          GT(UserCapabilityJunction.GRACE_PERIOD, 0),
```

Comment:
why does graceperiod need to be greater than 0 here?

Reply:
Yes, should be OR with GT(expiryPeriod   , GT(gracePeriod

Reply:
removed.